pkg/cmdutils/config.go

```diff
@@ -63,40 +49,68 @@ func (c *ClusterConfig) FlagSet() *pflag.FlagSet {
 		&c.WebServiceURL,
 		"admin-service-url",
 		"s",
-		pulsar.DefaultWebServiceURL,
+		c.WebServiceURL,
 		"The admin web service url that pulsarctl connects to.")
 
+	flags.StringVar(
+		&c.AuthPlugin,
+		"auth-plugin",
+		c.AuthPlugin,
+		"AuthPlugin is used to specify the plugin to use for authentication,\n"+
+			" the supported values are \"org.apache.pulsar.client.impl.auth.AuthenticationTls\"\n"+
+			" and \"org.apache.pulsar.client.impl.auth.AuthenticationToken\"")
+
 	flags.StringVar(
 		&c.AuthParams,
 		"auth-params",
-		"",
-		"Authentication parameters are used to configure the public and private key files required by tls\n"+
-			" For example: \"tlsCertFile:val1,tlsKeyFile:val2\"")
+		c.AuthParams,
+		"Authentication parameters are used to configure the authentication provider specified by"+
+			" \"AuthPlugin\".\n"+
+			" Tls example: \"tlsCertFile:val1,tlsKeyFile:val2\"\n"+
```

Comment:
fixed

Reply:
So my only hesitation is that auth-params is overloaded based on the value of the auth-plugin. For TLS Auth it would make sense to accept it as an array of parameters, but for Token Auth it is only ever a single string. Then there are the other authentication methods not implemented yet in pulsarctl, Athenz and Kerbos both expect it to be a JSON string.

I did add separate flags for "tls-key-file" and "tls-cert-file", to make it easier for TLS auth. I wonder if it would be best to omit "auth_plugin" and "auth_params" as flags and only use it through the env var config since it seems like those parameters were taken from the `client.conf` configuration.

Let me know your thoughts and I will make the necessary changes.

Reply:
make sense to me. I also just realized that your change didn't actually change the behavior here.